sdk/src/nonce_account.rs

```diff
@@ -21,13 +24,13 @@ pub fn create_account(lamports: u64) -> RefCell<AccountSharedData> {
 }
 
 // TODO: Consider changing argument from Hash to DurableNonce.
-pub fn verify_nonce_account(acc: &AccountSharedData, hash: &Hash) -> bool {
+pub fn verify_nonce_account(acc: &AccountSharedData, hash: &Hash) -> Option<Data> {
     if acc.owner() != &crate::system_program::id() {
-        return false;
+        return None;
     }
     match StateMut::<Versions>::state(acc).map(|v| v.convert_to_current()) {
-        Ok(State::Initialized(ref data)) => hash == &data.blockhash(),
-        _ => false,
+        Ok(State::Initialized(data)) => (hash == &data.blockhash()).then(|| data),
+        _ => None,
```

Comment:
Since you are changing this line, might be good to make this `_ => ... ` more explicit, and avoid bugs in the future if `State` ever has a 3rd variant. Something like:
```rust
Ok(State::Initialized(data)) => ...
Ok(State::Uninitialized) => None,
Err(_) => None,
```